### ISSUE TITLE
Improvements to be made to the GUI

### ISSUE THREAD

**Issue by ThomasLynn** [JMY1000/CyclesMineways#15]
The options that the user can modify are currently _in_ the script, this means it's easy for the user to not know that there are customisable features about the script (as what happened with @Masonrf).

One quick solution might be to have a popup display when the script is ran saying that it is customisable, though this will likely be annoying to a regular user, and they might not notice a new setting added to the script.

Another way of doing it is having the script create buttons. The user can then use these buttons to configure the script. Due to the buttons being in front of the user, it's unlikely they will not know about a feature.
After they have configured the script using these buttons, they can then press the "run script" button which will run the script.
This method has the advantage that all the options are given to the user, without them needing to know how to modify the script. It may also make it easier for the user, for example, tick boxes could be used to change USER_INPUT_SCENE.
This method of user interaction might be annoying, though. Because the Blender GUI is so cluttered with other stuff, and the user has full control over their own GUI, it's hard to find a menu to latch onto that the user will find easily (I recommend adding another tab to the left side of the viewports windows (along side "tools", "create", ect))

I've already created some GUI code, so it shouldn't be too hard to copy and paste that for other uses.

**Comment by Masonrf** [comment]
I would like to also suggest having images or a "video" file such as a gif on the wiki page on GitHub showing what the differences in the settings look like and link the corresponding page to the explanation of the setting on the GUI (I thought about having the images in the proposed GUI itself, but I don't know if its even possible). For some settings, like WATER_SHADER_TYPE, it would be nice to see the differences in the types before you run the script.

**Comment by ThomasLynn** [comment]
I've made a page on the wiki for water shaders
https://github.com/JMY1000/CyclesMineways/wiki/Water-Shader-Types
Maybe there could be a button in blender that opens that wiki page for reference?

Feel free to contribute to the wiki, if you want to.

**Comment by JMY1000** [comment]
Apologies for not being more active, I've been busy for the last three weeks working on a different project (different GitHub account for non-personal work.) I'll be working on maintaining this project better hopefully up until late August.

**Comment by ThomasLynn** [comment]
He's not dead guys!
![](https://66.media.tumblr.com/b7a581837daf8517e6a1a0a6a0295e64/tumblr_njja6tfNGZ1tffiako1_500.gif)

**Comment by Masonrf** [comment]
"Feel free to contribute to the wiki, if you want to."

Ahh, I have a bit to add, but I am unable to edit the wiki at all. How might I go about contributing to it?

**Comment by JMY1000** [comment]
@Masonrf You should now have the ability to edit all code, including the wiki.

**Comment by ThomasLynn** [comment]
@Masonrf There should be two buttons on the top-right of the scene saying "edit" and "new page". If there isn't, blame @JMY1000 :)

**Comment by JMY1000** [comment]
@ThomasLynn Only contributors have the ability to edit the wiki, so I made him a contributor.

**Comment by Masonrf** [comment]
^^

**Comment by Masonrf** [comment]
@JMY1000 do you want the GUI added to upcoming features?

**Comment by Masonrf** [comment]
Gosh darn the stupid "close and comment" button why do I always press you instead of just "comment"…

**Comment by JMY1000** [comment]
@Masonrf GUI is already in upcomingfeatures.txt

**Comment by JMY1000** [comment]
The more I look at what's been written for the GUI, the less I think it should be included with the *next upcoming* release. As useful as it is from the standpoint of a tech demo (seeing that we can implement these technologies into our code), in it's current form, it's simply not very useful from an end user standpoint and doesn't fit with how the script currently is setup to be used: set up options then execute, rather than execute each individual piece. While the second option in by no means invalid, and in fact could be quite useful often, the GUI clashes regardless.

In any case, I believe that the best course of action is to exclude what's been written from the next release. I think we should revert most changes made in [@c531072](https://github.com/JMY1000/CyclesMineways/commit/c531072d543e936afccc505cf786792db5ecee8d) for the main development branch and instead create a branch off of developer dedicated to working on the GUI. This will allow us to continue to push and release smaller changes without impacting our ability to work on the GUI. I think we should also take a similar approach to restructuring the water shader.

I haven't made these changes yet because I wanted to consult with the rest of the team regarding this.

**Comment by ThomasLynn** [comment]
I agree with you that it clashes. In its current state it's quite hard to find, and crashes a lot (I think it's a memory error due to me having a small rendered view port when I run the script). If we just remove it, push the script to master, then re-add it, that'll be fine.

I suggest you don't just quickly revert all the stuff done in c531072. We "could" leave the code in there, but comment out the part where it is assigned to Blender's GUI.
Changing def register and def unregister to pass, should work.
https://github.com/JMY1000/CyclesMineways/commit/c531072d543e936afccc505cf786792db5ecee8d#diff-eff090f38cbba489099d81d2fb2808a6R987
I haven't tested this, yet.

If we just comment out these lines, the code should work, but there will be junk code, which might confuse someone if they try to read the code.

**Comment by Masonrf** [comment]
I was already under the impression the GUI wasn't going to be this release. That's more of a "2.0" kind of version honestly. I agree with making a "Bleeding edge" fork, but two forks might be unnecessary. 

I'd say it should be fine to just comment it out and add a comment above each commented block about something along the lines of "This feature is a work in progress and will be released in a later version of the script."

**Comment by ThomasLynn** [comment]
That should be sufficient, yes. I'll go an comment stuff out now.
